### PR TITLE
Add Missing Argument on Call to InputOutputFile::open

### LINKED ISSUE #8229: Output Files Not Being Overwritten
Issue overview
--------------
The output files are not being overwritten by subsequent runs. I've dropped the terminal output from my testing here: https://gist.github.com/mitchute/fbebfdfbb092ea3a64c58e0d6d5f0702. 

To test, I took ```1ZoneUncontrolled.idf``` and ran it as a design-day only run. After tailing the CSV file, I forced an annual run and tailed the file again but am seeing the previous DD run results. Finally, I delete all of the eplus* files, run an annual simulation again, and tail the CSV file. Now the results line up with the annual run.

Maybe related to the recent PR #7904 ?

### Details
Some additional details for this issue (if relevant):
 - Platform (Mac 10.14.6)
 - Version of EnergyPlus (9.4.0 IOFreeze 908c5cedf5)
 - Unmethours link or helpdesk ticket number - None

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] ~Defect file added (list location of defect file here)~
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------
Fixes #8229 

InputOutputFile::open takes two arguments which both have default values. Only the second was being passed which was causing the ```forAppend``` flag to be set. 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by JasonGlazer** [comment]
I was seeing a related issue with the Initialization Summary in the tabular HTML file that was showing the tables with lots of duplicate rows. This was being caused by the EIO file being appended rather than overwritten each simulation. 

This branch fixed the issue with the Initialization Summary and EIO file.

File: src/EnergyPlus/IOFiles.cc [hunk 1]
```diff
@@ -177,7 +177,7 @@ void InputFile::backspace() noexcept
 InputOutputFile &InputOutputFile::ensure_open(const std::string &caller, bool output_to_file)
 {
     if (!good()) {
-        open(output_to_file);
+        open(false, output_to_file);
```
Comment (rraustad) [inline src/EnergyPlus/IOFiles.cc:180]: I am curious as to why this would compile before this change without the correct number of arguments?

Comment (rraustad) [reply src/EnergyPlus/IOFiles.cc:180]: Never mind, I see it in the override function.

**Comment by Myoldmopar** [comment]
This looks great.  No more issue with the API test.  I'm going to merge it.  Thanks @mitchute

**Comment by mjwitte** [comment]
@mitchute @Myoldmopar Somehow, this made the mtr files disappear.

**Comment by mitchute** [comment]
Interesting. How did this not get caught by CI?

**Comment by mjwitte** [comment]
I'm guessing if mathdiff doesn't find a file to compare, it doesn't show up in the CI diff log? I'm not sure what kind of message it puts out. I know that missing or added columns fall through silently, but it does try to compare any matching columns.

**Comment by mjwitte** [comment]
@mitchute Seems this change may not be the real culprit but somehow exposed issues with #7904. Discussing with @mbadams5.

**Pull request merged**